src/btree/bt_cursor.c

```diff
@@ -805,8 +805,6 @@ __wt_btcur_insert(WT_CURSOR_BTREE *cbt)
 
     /* It's no longer possible to bulk-load into the tree. */
     __wt_cursor_disable_bulk(session);
-    WT_ERR_ASSERT(
```

Comment:
The better change here is to switch to WT_RET_ASSERT, and I think it should go above the disable_bulk call - that's the code that assumes the session holds the "right" btree. Similarly below.

I'm happy to talk about when to choose ERR vs RET if that would help.

Reply:
No that makes sense.